### ISSUE TITLE
Settings / Providers / Status is not working

### ISSUE THREAD

**Issue by ngosang** [morpheus65535/bazarr#582]
**Describe the bug**
Subdivx provider is not working since few days (I will fix that). The issue is in the `Settings / Providers / Status` page, I see all providers are working and is misleading. You should show an error, warning or something.
![image](https://user-images.githubusercontent.com/10577978/65372540-0b91d580-dc72-11e9-8123-82f8f48a799b.png)

```
21/09/2019 13:06:45|INFO    |subliminal_patch.core           |Listing subtitles with provider 'subdivx' and languages set([<Language [es]>])|
21/09/2019 13:06:45|INFO    |subliminal_patch.core           |Initializing provider subdivx|
21/09/2019 13:06:45|DEBUG   |subliminal_patch.providers.subdivx|Searching subtitles u'Big Little Lies S02E05'|
21/09/2019 13:06:45|DEBUG   |urllib3.connectionpool          |Starting new HTTPS connection (1): www.subdivx.com:443|
21/09/2019 13:06:45|DEBUG   |subliminal_patch.http           |DNS: Falling back to default DNS or IP on www.subdivx.com|
21/09/2019 13:06:46|DEBUG   |urllib3.connectionpool          |https://www.subdivx.com:443 "GET /index.php?pg=1&accion=5&buscar=Big+Little+Lies+S02E05&oxdown=1 HTTP/1.1" 200 None|
21/09/2019 13:06:46|ERROR   |subliminal_patch.core           |Unexpected error in provider 'subdivx': Traceback (most recent call last):  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/core.py", line 162, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/providers/subdivx.py", line 173, in list_subtitles    episode=video.episode, year=video.year)  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/providers/subdivx.py", line 146, in query    ).find("a", {'target': 'new'})["href"].replace('http://', 'https://')TypeError: 'NoneType' object has no attribute '__getitem__'|Traceback (most recent call last):  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/core.py", line 162, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/providers/subdivx.py", line 173, in list_subtitles    episode=video.episode, year=video.year)  File "/root/scripts/bazarr/bazarr/../libs/subliminal_patch/providers/subdivx.py", line 146, in query    ).find("a", {'target': 'new'})["href"].replace('http://', 'https://')TypeError: 'NoneType' object has no attribute '__getitem__'|
21/09/2019 13:06:46|INFO    |subliminal_patch.core           |Discarding provider subdivx|
```

**Software (please complete the following information):**
 - Bazarr: 0.8.1

**Comment by halali** [comment]
The main issue is that, thus provider isn't well coded, there are missing raise exceptions, so bazarr is not able throttle this provider...

**Comment by halali** [comment]
So status is working for providers that support it

**Comment by morpheus65535** [comment]
@ngosang TY for the PR. Would it be possible to add the exception raise to make it complete?

**Comment by ngosang** [comment]
@halali @morpheus65535 give me an example of "well coded" provider to implement the exceptions.

**Comment by halali** [comment]
I do t want to say its bad coded, jus there is missing event Catchings somethink like in opensubtitles ```if status_code == 401:

raise Unauthorized

if status_code == 406:

raise NoSession

if status_code == 407:

raise DownloadLimitReached

if status_code == 413:

raise InvalidImdbid

if status_code == 414:

raise UnknownUserAgent

if status_code == 415:

raise DisabledUserAgent

if status_code == 429:

if not raise_api_limit:

raise TooManyRequests

else:

raise APIThrottled

if status_code == 503:

raise ServiceUnavailable(str(status_code))

if status_code != 200:

if response and "status" in response:

raise OpenSubtitlesError(response['status'])

raise ServiceUnavailable("Unknown Error, empty response: %s: %r" % (status_code, response))```

**Comment by morpheus65535** [comment]
https://github.com/morpheus65535/bazarr/blob/master/libs/subliminal/providers/opensubtitles.py

That’s a good exemple.

**Comment by ngosang** [comment]
@halali @morpheus65535 I did some tests, but it's not working.
Steps to reproduce:
1. Enable opensubtitles with wrong user/password
2. Perform a manual search
3. In debug level, you will see Unauthorized exception
4. In the Providers Status, opensubtitles is Good

NOTE: the same happens with automatic search

**Comment by halali** [comment]
There is place to improve, now throttling dont catch Unauthorized exception

**Comment by halali** [comment]
https://github.com/morpheus65535/bazarr/blob/master/bazarr/get_providers.py#L151
There is saving throttle to config file

**Comment by morpheus65535** [comment]
@halali Does this issue and related PR are under control? Do you take care of guiding @ngosang ? Thanks.

**Comment by morpheus65535** [comment]
Fixed by #585 and #586. Thanks @ngosang !
